catroid/src/androidTest/java/org/catrobat/catroid/test/formulaeditor/ParserTestOperators.java

```diff
@@ -336,4 +337,25 @@ public void testSubstraction() {
 		second.add(new InternToken(InternTokenType.NUMBER, "1"));
 		FormulaEditorTestUtil.testBinaryOperator(first, Operators.MINUS, second, Double.NaN, testSprite);
 	}
+
+	public void testNegativeZeroEqualsZero() {
+
+		FormulaElement formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, Integer.toString(-0), null), new FormulaElement(ElementType
+				.NUMBER, Integer.toString(0), null));
+		assertEquals("interpretOperatorEqual: -0 should be equal to 0", 1d, formulaElement.interpretRecursive(null));
+
+		formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, Integer.toString(0), null), new FormulaElement(ElementType.NUMBER,
+				Integer.toString(-0), null));
+		assertEquals("interpretOperatorEqual: 0 should be equal to -0", 1d, formulaElement.interpretRecursive(null));
+	}
+
+	public void testEqualsNaN() {
+		FormulaElement formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, String.valueOf(Double.NaN), null), new FormulaElement(ElementType
+				.NUMBER, String.valueOf(Double.NaN), null));
+		assertEquals("interpretOperatorEqual: NaN should be equal to NaN", 1d, formulaElement.interpretRecursive(null));
+	}
+
```

Comment:
If I don't put a new line then I get this checkstyle warning:-

:catroid:checkstyle[ant:checkstyle] [WARN] C:\Users\...\ParserTestOperators.java:0: File does not end with a newline. [NewlineAtEndOfFile]

but keeping this empty line removes such warnings.

Still, I will do the required changes.